src/pkcs15init/pkcs15-entersafe.c

```diff
@@ -130,6 +130,8 @@ static int entersafe_init_card(sc_profile_t *profile, sc_pkcs15_card_t *p15card)
 
 
 		 /* fill file by 0 */
+		 if (size > MAX_FILE_SIZE)
+			 LOG_TEST_RET(card->ctx, SC_ERROR_INVALID_DATA, "Initialize EF(DIR) failed with invalid file size");
```

Comment:
maybe better error explaining the invalid size is too large?

Reply:
Fixed